client/blocks/disconnect-jetpack-dialog/index.jsx

```diff
@@ -0,0 +1,141 @@
+import Dialog from 'components/dialog';
+import Button from 'components/button';
+import { recordGoogleEvent } from 'state/analytics/actions';
+
+class DisconnectJetpackDialog extends PureComponent {
+	trackReadMoreClick = () => {
+		this.props.recordGoogleEvent( 'Disconnect Jetpack Dialog', 'Clicked Read More about Jetpack benefits' );
+	};
+
+	translateArgs( icon ) {
+		return { components: { icon: <Gridicon icon={ icon } /> } };
+	}
+
+	planFeatures() {
+		const { plan, translate } = this.props;
+		const features = [];
+		switch ( plan ) {
+			case 'free':
+				features.push(
+					translate( '{{icon/}} Site stats, related content, and sharing tools',
+					this.translateArgs( 'stats-alt' )
+				) );
+				features.push(
+					translate( '{{icon/}} Brute force attack protection and uptime monitoring',
+					this.translateArgs( 'lock' )
+				) );
+				features.push( translate( '{{icon/}} Unlimited, high-speed image hosting', this.translateArgs( 'image' ) ) );
+				break;
+
+			case 'personal':
+				features.push( translate( '{{icon/}} Daily, automated backups (unlimited storage)', this.translateArgs( 'history' ) ) );
+				features.push( translate( '{{icon/}} Priority WordPress and security support', this.translateArgs( 'chat' ) ) );
+				features.push( translate( '{{icon/}} Spam filtering', this.translateArgs( 'spam' ) ) );
+				break;
+
+			case 'premium':
+				features.push( translate( '{{icon/}} Daily, automated backups (unlimited storage)', this.translateArgs( 'history' ) ) );
+				features.push( translate( '{{icon/}} Daily, automated malware scanning', this.translateArgs( 'spam' ) ) );
+				features.push( translate( '{{icon/}} Priority WordPress and security support', this.translateArgs( 'chat' ) ) );
+				features.push( translate( '{{icon/}} 13Gb of high-speed video hosting', this.translateArgs( 'video' ) ) );
+				break;
+
+			case 'professional':
+				features.push( translate( '{{icon/}} Daily, automated backups (unlimited storage)', this.translateArgs( 'history' ) ) );
+				features.push(
+					translate( '{{icon/}} Daily, automated malware scanning with automated resolution',
+					this.translateArgs( 'spam' )
+				) );
+				features.push( translate( '{{icon/}} Priority WordPress and security support', this.translateArgs( 'chat' ) ) );
+				features.push( translate( '{{icon/}} Unlimited high-speed video hosting', this.translateArgs( 'video' ) ) );
+				features.push( translate( '{{icon/}} SEO preview tools', this.translateArgs( 'globe' ) ) );
+				break;
+		}
+
+		return features.map( ( freature, index ) => {
+			return (
+				<div key={ 'disconnect-jetpack-dialog__feature-' + index } className="disconnect-jetpack-dialog__feature">
+					{ freature }
+				</div>
+			);
+		} );
+	}
+
+	render() {
+		const { onStay, onDisconnect, isVisible, translate, isBroken, siteName } = this.props;
+		if ( isBroken ) {
+			return (
+				<Dialog isVisible={ isVisible } additionalClassNames="disconnect-jetpack-dialog" onClose={ onStay } >
```

Comment:
Should we extract common things from the broken/not broken cases to another component/method?